scripts/eval_trajectory_to_lisdf.py

```diff
@@ -56,16 +59,76 @@ def _main() -> None:
+    gripper_idxs = [robot.left_finger_joint_idx, robot.right_finger_joint_idx]
+    arm_joint_names = list(np.delete(joint_names, gripper_idxs))
+
+    def _joints_to_arm_gripper_vals(
+        joints: NDArray[np.float32]
+    ) -> Tuple[NDArray[np.float32], GripperPosition]:
+        # Determine if gripper is currently closed or open.
+        # Use arbitrary gripper idx.
+        gripper_val = joints[gripper_idxs[0]]
+        dist_to_open = abs(gripper_val - robot.open_fingers)
+        dist_to_closed = abs(gripper_val - robot.closed_fingers)
+        if dist_to_closed < dist_to_open:
+            gripper_state = GripperPosition.close
+        else:
+            gripper_state = GripperPosition.open
+        # Extract the arm joint state.
+        arm_state = np.delete(joints, gripper_idxs)
+        return (arm_state, gripper_state)
+
+    arm_gripper_states = list(map(_joints_to_arm_gripper_vals, joint_arr))
+
+    # List of LISDF commands, alternating JointSpacePath and ActuateGripper.
+    commands: List[Command] = []
+
+    # Accumulate the arm-only actions until a change in the gripper state
+    # is observed, at which point we finalize the JointSpacePath command,
+    # add an ActuateGripper command, and then start a new JointSpacePath.
+    accum_arm_states: List[NDArray[np.float32]] = []
+
+    # Helper function for converting an accum_arm_states to a JointSpacePath.
+    def _create_path_command(
+            arm_states: List[NDArray[np.float32]]) -> JointSpacePath:
+        duration = args.time_per_conf * len(arm_states)
+        arm_states_np = np.array(arm_states)
+        path_command = JointSpacePath.from_waypoints_np_array(
+            arm_states_np,
+            arm_joint_names,
+            duration=duration,
+            label=args.input,
+        )
+        return path_command
+
+    # Track changes in the gripper state.
+    _, prev_gripper_state = arm_gripper_states[0]
+
+    for arm_state, current_gripper_state in arm_gripper_states:
+        accum_arm_states.append(arm_state)
+        # Check if a change in the gripper state has occurred.
+        if current_gripper_state != prev_gripper_state:
+            # Finalize the JointSpacePath until this point.
+            path_command = _create_path_command(accum_arm_states)
+            commands.append(path_command)
+            accum_arm_states = [arm_state]
+            # Create the ActuateGripper command.
+            gripper_command = ActuateGripper(
+                configurations={"panda_gripper": current_gripper_state},
+                label="gripper_action")
+            commands.append(gripper_command)
+            prev_gripper_state = current_gripper_state
+
+    # Finish the last JointSpacePath.
+    if len(accum_arm_states) > 1:
+        path_command = _create_path_command(accum_arm_states)
+        commands.append(path_command)
+
+    # Write out the commands to JSON.
+    lisdf_plan = LISDFPlan(commands=commands, lisdf_problem="no_problem")
```

Comment:
Maybe let's just set the `lisdf_problem` to be the filename of the saved trajectory just so we can track what is what.